bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -837,7 +840,7 @@ void flushRotatedLogs() throws IOException {
         leastUnflushedLogId = flushedLogId + 1;
     }
 
-    void flush() throws IOException {
+    public synchronized void flush() throws IOException {
```

Comment:
any reason why `synchronized` the whole method?

Reply:
There can only be a single `flush()` happening at a point in time. That will not affect the `addEntry()` which is only stopped while the 2 `WriteCache` instances are swapped.

Reply:
Uhm, I thought this was `LedgerStorage.flush()`. Let me revisit this.

Reply:
Yes, that was not needed indeed.